### ISSUE TITLE
golint suggests removing else illegally even when it is necessary.

### ISSUE THREAD

**Issue by ymmt2005** [golang/lint#202]
golint warns for code like this:

```go
    if v, ok := aMap[key]; !ok {
        return
    } else {
        // code use v
    }
```

The warning says "if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)", but of course `else` cannot be removed here.

**Comment by dominikh** [comment]
> move short variable declaration to its own line if necessary

**Comment by ymmt2005** [comment]
Thank you.

> move short variable declaration to its own line if necessary

Doing so will enlarge the variable scope, which may break codes.

I think it is arguable which is better:
* to outdent a block in exchange for wider scope, or
* to narrow variable scopes in exchange for an else block.

In case `else` block has just 1 or 2 lines of code, having
short variable declaration and `else` block is not so bad.

**Comment by dsymonds** [comment]
The established Go style is what golint suggests, even if that isn't your preference.

**Comment by ymmt2005** [comment]
@dominikh @dsymonds 
Just to clarify, I can *fix* the code as follows:

```go
    if v, ok := aMap[key]; ok {
        // code use v
    } else {
        return
    }
```

But I don't understand why this is OK for golint while the first is not.
Is this intended behavior?

A real-world example is here.
https://github.com/cybozu-go/goma/commit/cfded2d3d7bcf0f56fbb02836300b838da55c371#diff-ad79469688b87932daa9a5969430cc26

**Comment by dsymonds** [comment]
Golint doesn't try to catch every possible situation. Your code isn't "broken", it just doesn't follow the standard Go style, which would be to write

```go
v, ok := aMap[key]
if !ok {
  return
}
// use v
```

**Comment by ymmt2005** [comment]
@dsymonds 
I see.  Thank you!
